### PR TITLE
Plug shibuqam and an other auth

### PR DESCRIPTION
**Pull request by privat**
This detach github a lot more and allow to plug more authentications schema.
A shibuqamauth and a random auth are provided.

I do not know hot to handle the various auth method from a UI point of view.
Currently, I just hijacked the `AuthRouter` to add specific login subroutes.

http://localhost:3000/auth/login is still the Github login
http://localhost:3000/auth/shiblogin is the shibuqam login
http://localhost:3000/auth/loginas?id=Morriar is the random login

### CODE REVIEW

**Comment by Morriar** [comment]
test this please

File: Makefile [hunk 1]
```diff
@@ -15,7 +15,7 @@
 build:
 	mkdir -p bin
 	#nitserial src/app.nit -o src/app_serial.nit
-	nitc src/app.nit -m src/app_serial.nit -o bin/app
+	nitc src/app_debug.nit -m src/app_serial.nit -o bin/app
```
Comment (Morriar) [inline Makefile:22]: woot?

Comment (privat) [inline Makefile:22]: I wanted to distinguish a debug version with auth_rand and a pulished one

File: src/api/api_auth_rand.nit [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+import api_auth
+
+redef class AuthRouter
+	redef init do
+		use("/loginas", new RandLogin(config))
+	end
+end
+
+class RandLogin
```
Comment (Morriar) [inline src/api/api_auth_rand.nit:30]: You should give an example of the uri to use

Comment (Morriar) [inline src/api/api_auth_rand.nit:30]: ... with the get args

File: Makefile (hunk 1 above)
Comment (Morriar) [inline Makefile:22]: but why destroy my magnificient `build` goal? The server needs it

Comment (Morriar) [inline Makefile:22]: Can't you add a `debug` goal?

**Comment by privat** [comment]
reroll: apply remarke and improve the new shibuqamoauth module

**Comment by privat** [comment]
rebase: try be be compatible with the lastest morriar/nit/next

**Comment by ppepos** [comment]
+1

**Pull request merged**